src/main/java/seedu/toluist/model/Task.java

```diff
@@ -83,9 +89,11 @@ public boolean equals(Object other) {
                 && this.description.equals(((Task) other).description)) // state check
                 && this.priority.equals(((Task) other).priority)
                 && this.allTags.equals(((Task) other).allTags)
+                && Objects.equals(this.recurringFrequency, ((Task) other).recurringFrequency) // handles null
```

Comment:
might be quite complicated if u want to handle all these, and out of the scope for this branch i think.